sql/core/src/test/scala/org/apache/spark/sql/CollationSuite.scala

```diff
@@ -640,6 +641,196 @@ class CollationSuite extends DatasourceV2SQLBase with AdaptiveSparkPlanHelper {
         "reason" -> "generation expression cannot contain non-default collated string type"))
   }
 
+  trait ArrayCheck {
+    def dataType: String
+    def dataTypeCollated: String
+  }
+
+  trait ArrayCheckSimple extends ArrayCheck {
+    override def dataType: String = s"array<string>"
```

Comment:
nit: let's remove those.

```suggestion
    override def dataType: String = "array<string>"
```